primitives/runtime/src/lib.rs

```diff
@@ -95,7 +95,57 @@ pub use either::Either;
 /// the block itself would allow swapping justifications to change the block's hash
 /// (and thus fork the chain). Sending a `Justification` alongside a block instead
 /// bypasses this problem.
-pub type Justification = Vec<u8>;
+pub type EncodedJustification = Vec<u8>;
+
+/// Justification together with an ID to tag its origin.
```

Comment:
It would make more sense to have the longer comment above here and extend it with the details about the consensus engine id.